src/overlays/actors/ovl_En_Xc/z_en_xc.c

```diff
@@ -77,479 +47,2365 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    // Player is adult and does not have iron boots and has not learned Serenade
+    if ((!CHECK_OWNED_EQUIP(EQUIP_BOOTS, 1) && !(gSaveContext.eventChkInf[5] & 4)) && LINK_IS_ADULT) {
+        this->action = SHIEK_ACTION_SERENADE;
+        osSyncPrintf("水のセレナーデ シーク誕生!!!!!!!!!!!!!!!!!!\n");
+    } else {
+        Actor_Kill(&this->actor);
+        osSyncPrintf("水のセレナーデ シーク消滅!!!!!!!!!!!!!!!!!!\n");
+    }
+}
+
+s32 EnXc_SerenadeCS(EnXc* this, GlobalContext* globalCtx) {
+    if (this->actor.params == SHIEK_TYPE_SERENADE) {
+        Player* player = PLAYER;
+        s32 stateFlags = player->stateFlags1;
+        if (CHECK_OWNED_EQUIP(EQUIP_BOOTS, 1) && !(gSaveContext.eventChkInf[5] & 4) && ((stateFlags << 2) >= 0) &&
+            !Gameplay_InCsMode(globalCtx)) {
+            Cutscene_SetSegment(globalCtx, &ice_doukutu_sceneCutsceneData0x000330);
+            gSaveContext.cutsceneTrigger = 1;
+            gSaveContext.eventChkInf[5] |= 4; // Learned Serenade of Water Flag
+            Item_Give(globalCtx, ITEM_SONG_SERENADE);
+            osSyncPrintf("ブーツを取った!!!!!!!!!!!!!!!!!!\n");
+            return true;
+        }
+        osSyncPrintf("はやくブーツを取るべし!!!!!!!!!!!!!!!!!!\n");
+        return false;
+    }
+    return true;
+}
+
+void EnXc_DoNothing(EnXc* this, GlobalContext* globalCtx) {
+}
+
+void EnXc_SetWalkingSFX(EnXc* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    u32 sfxId;
+    s32 pad2;
+
+    if ((Animation_OnFrame(&this->skelAnime, 11.0f)) || (Animation_OnFrame(&this->skelAnime, 23.0f))) {
+        if (this->actor.bgCheckFlags & 1) {
+            sfxId = SFX_FLAG;
+            sfxId += SurfaceType_GetSfx(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId);
+            func_80078914(&this->actor.projectedPos, sfxId);
+        }
+    }
+}
+
+void EnXc_SetNutThrowSFX(EnXc* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    u32 sfxId;
+    s32 pad2;
+
+    if (Animation_OnFrame(&this->skelAnime, 7.0f)) {
+        if (this->actor.bgCheckFlags & 1) {
+            sfxId = SFX_FLAG;
+            sfxId += SurfaceType_GetSfx(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId);
+            func_80078914(&this->actor.projectedPos, sfxId);
+        }
+    }
+    if (Animation_OnFrame(&this->skelAnime, 20.0f)) {
+        func_80078914(&this->actor.projectedPos, NA_SE_VO_SK_SHOUT);
+    }
+}
+
+void EnXc_SetLandingSFX(EnXc* this, GlobalContext* globalCtx) {
+    u32 sfxId;
+    s16 sceneNum = globalCtx->sceneNum;
+
+    if ((gSaveContext.sceneSetupIndex != 4) || (sceneNum != SCENE_SPOT11)) {
+        if (Animation_OnFrame(&this->skelAnime, 11.0f)) {
+            sfxId = SFX_FLAG;
+            sfxId += SurfaceType_GetSfx(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId);
+            func_80078914(&this->actor.projectedPos, sfxId);
+        }
+    }
+}
+
+void EnXc_SetColossusAppearSFX(EnXc* this, GlobalContext* globalCtx) {
+    s16 sceneNum;
+    static Vec3f sXyzDist;
```

Comment:
Relevant in other places in this file
```suggestion
    static Vec3f xyzDist;
    s16 sceneNum;
```